src/texture.rs

```diff
@@ -69,3 +70,60 @@ impl<T> Texture<T> {
         ]
     }
 }
+
+/// Represents paths to cube map texture, useful for loading
+/// [`CubeMap`](struct.CubeMap.html).
+#[derive(Clone, Debug)]
+pub struct CubeMapPath<P: AsRef<Path>> {
```

Comment:
Having a single `P: AsRef<Path>` type argument means every path source must be the same type.

```rust
let path = CubeMapPath {
    front: "front.png",
    back: Path::new("back.png"),
    // etc.
};
```
would fail to compile.

Reply:
I know. Do you have any proposal? I decided to do it this way because it's odd to use different path types for single cubemap

Reply:
There are a few options:

1. Explicit type for every param
```rust
pub struct CubeMapPath<F, B, L, R, U, D>
where
    F: AsRef<Path>,
    B: AsRef<Path>,
    L: AsRef<Path>,
    R: AsRef<Path>,
    U: AsRef<Path>,
    D: AsRef<Path>,
{
     /// "Front" image. `Z+`.
    pub front: F,
    /// "Back" image. `Z-`.
    pub back: B,
    /// "Left" image. `X-`.
    pub left: L,
    /// "Right" image. `X+`.
    pub right: R,
    /// "Up" image. `Y+`.
    pub up: U,
    /// "Down" image. `Y-`.
    pub down: D,
}
```
2. Use dynamic dispatch.
```rust
pub struct CubeMapPath<'a> {
     /// "Front" image. `Z+`.
    pub front: &'a AsRef<Path>,
    /// "Back" image. `Z-`.
    pub back: &'a AsRef<Path>,
    /// "Left" image. `X-`.
    pub left: &'a AsRef<Path>,
    /// "Right" image. `X+`.
    pub right: &'a AsRef<Path>,
    /// "Up" image. `Y+`.
    pub up: &'a AsRef<Path>,
    /// "Down" image. `Y-`.
    pub down: &'a AsRef<Path>,
}
```
3. Use concrete type.
```rust
pub struct CubeMapPath<'a> {
     /// "Front" image. `Z+`.
    pub front: &'a Path,
    /// "Back" image. `Z-`.
    pub back: &'a Path,
    /// "Left" image. `X-`.
    pub left: &'a Path,
    /// "Right" image. `X+`.
    pub right: &'a Path,
    /// "Up" image. `Y+`.
    pub up: &'a Path,
    /// "Down" image. `Y-`.
    pub down: &'a Path,
}
```

Reply:
Option 1 is zero-cost and most flexible, although ugly.

Reply:
First one is too ugly (imagine if the user wants to store that inside some struct).
I don't know, I would like to avoid such complexity and have only one type, user can convert if he wants.

Reply:
I think it's fine for all the them to be same type here